README.md
# Install
```
npm i @nfq/safe-validation-regex
```

# Usage
```javascript
import {url} from '@nfq/safe-validation-regex';

const checkUrl = (uri) => {
    return url.test(uri);
}
```

# Documentation
The following regexes are integrated for now:

## **url**

`/^((((https?|ftps?|gopher|telnet|nntp):\/\/)|(mailto:|news:))(%[0-9A-Fa-f]{2}|[-()_.!~*';/?:@&=+$,A-Za-z0-9])+)([).!';/?:,][[:blank:|:blank:]])?$`

A valid URL per the URL spec.

---

## **ip**

`^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$`

A valid IP Address

---

## **email**

`^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,7}$`

A valid e-mail address

---

## **safeText**

`^[a-zA-Z0-9 .-]+$`

Lower and upper case letters and all digits

---

## **usDate**

`^(?:(?:(?:0?[13578]|1[02])(\/|-|\.)31)\1|(?:(?:0?[1,3-9]|1[0-2])(\/|-|\.)(?:29|30)\2))(?:(?:1[6-9]|[2-9]\d)?\d{2})$|^(?:0?2(\/|-|\.)29\3(?:(?:(?:1[6-9]|[2-9]\d)?(?:0[48]|[2468][048]|[13579][26])|(?:(?:16|[2468][048]|[3579][26])00))))$|^(?:(?:0?[1-9])|(?:1[0-2]))(\/|-|\.)(?:0?[1-9]|1\d|2[0-8])\4(?:(?:1[6-9]|[2-9]\d)?\d{2})$`

Date in US format with support for leap years

---

## **deDate**

`^((((0?[1-9]|[12]\d|3[01])[\.\-\/](0?[13578]|1[02])[\.\-\/]((1[6-9]|[2-9]\d)?\d{2}))|((0?[1-9]|[12]\d|30)[\.\-\/](0?[13456789]|1[012])[\.\-\/]((1[6-9]|[2-9]\d)?\d{2}))|((0?[1-9]|1\d|2[0-8])[\.\-\/]0?2[\.\-\/]((1[6-9]|[2-9]\d)?\d{2}))|(29[\.\-\/]0?2[\.\-\/]((1[6-9]|[2-9]\d)?(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00)|00)))|(((0[1-9]|[12]\d|3[01])(0[13578]|1[02])((1[6-9]|[2-9]\d)?\d{2}))|((0[1-9]|[12]\d|30)(0[13456789]|1[012])((1[6-9]|[2-9]\d)?\d{2}))|((0[1-9]|1\d|2[0-8])02((1[6-9]|[2-9]\d)?\d{2}))|(2902((1[6-9]|[2-9]\d)?(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00)|00))))$`

Date in German format with support for leap years

---

## **creditcard**

`^((4\d{3})|(5[1-5]\d{2})|(6011)|(7\d{3}))-?\d{4}-?\d{4}-?\d{4}|3[4,7]\d{13}$`

A valid credit card number

---

## **simplePassword**

`^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{4,8}$`

4 to 8 character password requiring numbers and both lowercase and uppercase letters

---

## **complexPassword**

`^(?:(?=.*\d)(?=.*[A-Z])(?=.*[a-z])|(?=.*\d)(?=.*[^A-Za-z0-9])(?=.*[a-z])|(?=.*[^A-Za-z0-9])(?=.*[A-Z])(?=.*[a-z])|(?=.*\d)(?=.*[A-Z])(?=.*[^A-Za-z0-9]))(?!.*(.)\1{2,})[A-Za-z0-9!~<>,;:_=?*+#."&§%°()\|\[\]\-\$\^\@\/]{8,32}$`

4 to 32 character password requiring at least 3 out 4 (uppercase and lowercase letters, numbers and special characters) and no more than 2 equal characters in a row

---

## **englishDigitWords**

`^(zero|one|two|three|four|five|six|seven|eight|nine)$`

The English words representing the digits 0 to 9

---

## **germanDigitWords**

`^(null|eins|zwei|drei|vier|f(ue|ü)nf|sechs|sieben|acht|neun)$`

The German words representing the digits 0 to 9

---

## **englishDayWords**

`^(Mo|Tu|We|Th|Fr|Sa|Su)$`

English 2 character abbreviations for the days of the week

---

## **germanDayWords**

`^(Mo|Di|Mi|Do|Fr|Sa|So)$`

German 2 character abbreviations for the days of the week

---

## **englishMonthWords**

`^(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)$`

English 3 character abbreviations for the months

---

## **germanMonthWords**

`^(Jan|Feb|M(ae|ä)r|Apr|Mai|Jun|Jul|Aug|Sep|Oct|Nov|Dec)$`

German 3 character abbreviations for the months

---

## **usZipCode**

`^\d{5}(-\d{4})?$`

US zip code with optional dash-four

---

## **germanZipCode**

`^\d{5}$`

DE zip code with optional dash-four

---

## **cssColorCode**

`^#?([a-f]|[A-F]|[0-9]){3}(([a-f]|[A-F]|[0-9]){3})?$`

CSS hex colorcode

---

## **intNumber**

`^[-+]?[0-9]+$`

Integer numbers

---

## **floatNumber**

`^[-+]?[0-9]+[.]?[0-9]*([eE][-+]?[0-9]+)?$`

Floating point numbers

---

## **personName**

`^[a-zA-Z]+(([',. -][a-zA-Z ])?[a-zA-Z]*)*$`

Person names.

---

## **macAdress**

`^([0-9a-fA-F][0-9a-fA-F]:){5}([0-9a-fA-F][0-9a-fA-F])$`

Valid MAC adress after specification

---

## **guid**

`^[A-Z0-9]{8}-[A-Z0-9]{4}-[A-Z0-9]{4}-[A-Z0-9]{4}-[A-Z0-9]{12}$`

Valid GUID

---

## **domainName**

`^([a-zA-Z0-9]([a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]{2,6}$`

Reasonable domain name

---

## **windowsFileName**

`^(?!^(PRN|AUX|CLOCK\$|NUL|CON|COM\d|LPT\d|\..*)(\..+)?$)[^\x00-\x1f\\?*:\";|/]+$`

Valid Windows filename

---

## **anyPlattformFileName**

`^(([a-zA-Z]:|\\)\\)?(((\.)|(\.\.)|([^\\/:*?"|<>. ](([^\\/:*?"|<>. ])|([^\\/:*?"|<>]*[^\\/:*?"|<>. ]))?))\\)*[^\\/:*?"|<>. ](([^\\/:*?"|<>. ])|([^\\/:*?"|<>]*[^\\/:*?"|<>. ]))?$`

Valid filename for any plattform

CHANGELOG.md
# Changelog

All notable changes to this project will be documented in this file. See [standard-version](https://github.com/conventional-changelog/standard-version) for commit guidelines.

### [1.4.1](https://github.com/nfqde/nfq-safe-validation-regexes/compare/v1.4.0...v1.4.1) (2023-08-22)


### Bug Fixes

* **Packages:** Update packages ([#13](https://github.com/nfqde/nfq-safe-validation-regexes/issues/13)) ([7c3ec64](https://github.com/nfqde/nfq-safe-validation-regexes/commit/7c3ec64602e9feb0793162e71a2f223458ab2336))

## [1.4.0](https://github.com/nfqde/nfq-safe-validation-regexes/compare/v1.3.3...v1.4.0) (2023-06-20)


### Features

* **App:** Update packages ([4e5d06f](https://github.com/nfqde/nfq-safe-validation-regexes/commit/4e5d06f38575982c454f11a323356fdda6c725f3))

### [1.3.3](https://github.com/nfqde/nfq-safe-validation-regexes/compare/v1.3.2...v1.3.3) (2023-05-02)


### Bug Fixes

* **Deps:** up deps ([#11](https://github.com/nfqde/nfq-safe-validation-regexes/issues/11)) ([7682224](https://github.com/nfqde/nfq-safe-validation-regexes/commit/768222479d964b5ae275882475abe2a14a0674a1))

### [1.3.2](https://github.com/nfqde/nfq-safe-validation-regexes/compare/v1.3.1...v1.3.2) (2023-03-05)


### Bug Fixes

* **Month:** Fix month regex ([#10](https://github.com/nfqde/nfq-safe-validation-regexes/issues/10)) ([721acb2](https://github.com/nfqde/nfq-safe-validation-regexes/commit/721acb298fe6d8d045bee3013d1c8b8b472f0672)), closes [#6](https://github.com/nfqde/nfq-safe-validation-regexes/issues/6)

### [1.3.1](https://github.com/nfqde/nfq-safe-validation-regexes/compare/v1.3.0...v1.3.1) (2023-03-05)


### Bug Fixes

* **Ignore:** Ignore cypress config ([#9](https://github.com/nfqde/nfq-safe-validation-regexes/issues/9)) ([6139b4d](https://github.com/nfqde/nfq-safe-validation-regexes/commit/6139b4d75b9b6d8f48b30384e4da71fcea58d9e0))

## [1.3.0](https://github.com/nfqde/nfq-safe-validation-regexes/compare/v1.2.0...v1.3.0) (2023-03-05)


### Features

* **Typescript:** Full rebuild with typescript ([#7](https://github.com/nfqde/nfq-safe-validation-regexes/issues/7)) ([62a39b4](https://github.com/nfqde/nfq-safe-validation-regexes/commit/62a39b482f61605771e545f76a9377484185493f))

## [1.2.0](https://github.com/nfqde/nfq-safe-validation-regexes/compare/v1.1.0...v1.2.0) (2021-10-17)


### Features

* **Text:** Allow umlaut characters ([#3](https://github.com/nfqde/nfq-safe-validation-regexes/issues/3)) ([26ca0bb](https://github.com/nfqde/nfq-safe-validation-regexes/commit/26ca0bb680018a0490bd807c2fc2f1f9fc82528b))

## [1.1.0](https://github.com/nfqde/nfq-safe-validation-regexes/compare/v1.0.0...v1.1.0) (2020-05-15)


### Features

* **Regex:** Add integer regex ([#2](https://github.com/nfqde/nfq-safe-validation-regexes/issues/2)) ([6a0dea7](https://github.com/nfqde/nfq-safe-validation-regexes/commit/6a0dea77294ec8dfc95456dd6bc2aa55baaeacaf))

## 1.0.0 (2020-05-12)


src/index.js
const url = /^((((https?|ftps?|gopher|telnet|nntp):\/\/)|(mailto:|news:))(%[0-9A-Fa-f]{2}|[-()_.!~*';/?:@&=+$,A-Za-z0-9])+)([).!';/?:,][[:blank:|:blank:]])?$/;
const ip = /^(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)\.(25[0-5]|2[0-4][0-9]|[01]?[0-9][0-9]?)$/;
const email = /^[a-zA-Z0-9_+&*-]+(?:\.[a-zA-Z0-9_+&*-]+)*@(?:[a-zA-Z0-9-]+\.)+[a-zA-Z]{2,7}$/;
const safeText = /^[a-zA-Z0-9 .-\u00F0-\u02AF]+$/;
const usDate = /^(?:(?:(?:0?[13578]|1[02])(\/|-|\.)31)\1|(?:(?:0?[1,3-9]|1[0-2])(\/|-|\.)(?:29|30)\2))(?:(?:1[6-9]|[2-9]\d)?\d{2})$|^(?:0?2(\/|-|\.)29\3(?:(?:(?:1[6-9]|[2-9]\d)?(?:0[48]|[2468][048]|[13579][26])|(?:(?:16|[2468][048]|[3579][26])00))))$|^(?:(?:0?[1-9])|(?:1[0-2]))(\/|-|\.)(?:0?[1-9]|1\d|2[0-8])\4(?:(?:1[6-9]|[2-9]\d)?\d{2})$/;
const deDate = /^((((0?[1-9]|[12]\d|3[01])[\.\-\/](0?[13578]|1[02])[\.\-\/]((1[6-9]|[2-9]\d)?\d{2}))|((0?[1-9]|[12]\d|30)[\.\-\/](0?[13456789]|1[012])[\.\-\/]((1[6-9]|[2-9]\d)?\d{2}))|((0?[1-9]|1\d|2[0-8])[\.\-\/]0?2[\.\-\/]((1[6-9]|[2-9]\d)?\d{2}))|(29[\.\-\/]0?2[\.\-\/]((1[6-9]|[2-9]\d)?(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00)|00)))|(((0[1-9]|[12]\d|3[01])(0[13578]|1[02])((1[6-9]|[2-9]\d)?\d{2}))|((0[1-9]|[12]\d|30)(0[13456789]|1[012])((1[6-9]|[2-9]\d)?\d{2}))|((0[1-9]|1\d|2[0-8])02((1[6-9]|[2-9]\d)?\d{2}))|(2902((1[6-9]|[2-9]\d)?(0[48]|[2468][048]|[13579][26])|((16|[2468][048]|[3579][26])00)|00))))$/;
const creditcard = /^((4\d{3})|(5[1-5]\d{2})|(6011)|(7\d{3}))-?\d{4}-?\d{4}-?\d{4}|3[4,7]\d{13}$/;
const simplePassword = /^(?=.*\d)(?=.*[a-z])(?=.*[A-Z]).{4,8}$/; // 4 to 8 chars, lower and uppercase
const complexPassword = /^(?:(?=.*\d)(?=.*[A-Z])(?=.*[a-z])|(?=.*\d)(?=.*[^A-Za-z0-9])(?=.*[a-z])|(?=.*[^A-Za-z0-9])(?=.*[A-Z])(?=.*[a-z])|(?=.*\d)(?=.*[A-Z])(?=.*[^A-Za-z0-9]))(?!.*(.)\1{2,})[A-Za-z0-9!~<>,;:_=?*+#."&§%°()\|\[\]\-\$\^\@\/]{8,32}$/; // 4 to 32 chars, lower-, uppercase, numbers, special chars. Max 2 same chars in row.
const englishDigitWords = /^(zero|one|two|three|four|five|six|seven|eight|nine)$/;
const germanDigitWords = /^(null|eins|zwei|drei|vier|f(ue|ü)nf|sechs|sieben|acht|neun)$/;
const englishDayWords = /^(Mo|Tu|We|Th|Fr|Sa|Su)$/;
const germanDayWords = /^(Mo|Di|Mi|Do|Fr|Sa|So)$/;
const englishMonthWords = /^(Jan|Feb|Mar|Apr|May|Jun|Jul|Aug|Sep|Oct|Nov|Dec)$/;
const germanMonthWords = /^(Jan|Feb|M(ae|ä)r|Apr|Mai|Jun|Jul|Aug|Sep|Oct|Nov|Dec)$/;
const usZipCode = /^\d{5}(-\d{4})?$/;
const germanZipCode = /^\d{5}$/;
const cssColorCode = /^#?([a-f]|[A-F]|[0-9]){3}(([a-f]|[A-F]|[0-9]){3})?$/;
const intNumber = /^[-+]?[0-9]+$/;
const floatNumber = /^[-+]?[0-9]+[.]?[0-9]*([eE][-+]?[0-9]+)?$/;
const personName = /^[a-zA-Z\u00F0-\u02AF]+(([',. -][a-zA-Z\u00F0-\u02AF ])?[a-zA-Z\u00F0-\u02AF]*)*$/;
const macAdress = /^([0-9a-fA-F][0-9a-fA-F]:){5}([0-9a-fA-F][0-9a-fA-F])$/;
const guid = /^[A-Z0-9]{8}-[A-Z0-9]{4}-[A-Z0-9]{4}-[A-Z0-9]{4}-[A-Z0-9]{12}$/;
const domainName = /^([a-zA-Z0-9]([a-zA-Z0-9\-]{0,61}[a-zA-Z0-9])?\.)+[a-zA-Z]{2,6}$/;
const windowsFileName = /^(?!^(PRN|AUX|CLOCK\$|NUL|CON|COM\d|LPT\d|\..*)(\..+)?$)[^\x00-\x1f\\?*:\";|/]+$/;
const anyPlattformFileName = /^(([a-zA-Z]:|\\)\\)?(((\.)|(\.\.)|([^\\/:*?"|<>. ](([^\\/:*?"|<>. ])|([^\\/:*?"|<>]*[^\\/:*?"|<>. ]))?))\\)*[^\\/:*?"|<>. ](([^\\/:*?"|<>. ])|([^\\/:*?"|<>]*[^\\/:*?"|<>. ]))?$/;

export {
    url,
    ip,
    email,
    safeText,
    usDate,
    deDate,
    creditcard,
    simplePassword,
    complexPassword,
    englishDigitWords,
    germanDigitWords,
    englishDayWords,
    germanDayWords,
    englishMonthWords,
    germanMonthWords,
    usZipCode,
    germanZipCode,
    cssColorCode,
    intNumber,
    floatNumber,
    personName,
    macAdress,
    guid,
    domainName,
    windowsFileName,
    anyPlattformFileName
};